warehouse/forklift/legacy.py

```diff
@@ -1150,6 +1150,7 @@ def file_upload(request):
                 )
                 if request.oidc_publisher
                 else None,
+                "uploaded_via_trusted_publisher": bool(request.oidc_publisher),
```

Comment:
Yeah, this isn't fantastic (and is the problem with JSON types in DBs -- it's too tempting to put yet another key in!). I don't have a better idea here, unfortunately, except maybe to offload this onto the `File` model itself.

Thoughts on that? In that case, we'd have something like `File.trusted_published` which we could both initialize for new uploads and backfill from events for old ones.

Reply:
Agreed - this was definitely a good placeholder while we learned about the utility of the data and how we would use it. `File.uploaded_via_trusted_publisher` grew out of me wanting to explore how it might be used - currently only in Admin UI via [`Release.trusted_published`](https://github.com/pypi/warehouse/blob/5466d3a4ae6f1b2d6c81f7ba91d313175d1d0233/warehouse/packaging/models.py#L636-L645).

A bool lookup ought to be faster, for sure.

I'm happy to proceed with this merge and get this going, if you wanted to work through a data migration and refactor apart, so we can benefit from these changes already. Your call!

Reply:
> I'm happy to proceed with this merge and get this going, if you wanted to work through a data migration and refactor apart, so we can benefit from these changes already. Your call!

Yep, that SGTM! I can work on that after this is merged.